mobile/lib/screens/home_screen_router.dart

```diff
@@ -294,6 +244,16 @@ class _HomeScreenRouterState extends ConsumerState<HomeScreenRouter>
               }
             }
 
+            // Pre-initialize controllers for adjacent videos on initial build
+            WidgetsBinding.instance.addPostFrameCallback((_) {
```

Comment:
We shouldn't call a post frame callback inside the build method, or do anything that would cause a side effect inside the build method.

I understand that this call is inside the `onData`, but that is still inside the build method and we don't have control over when this will be called.

If we want to run something on the initial build, we should do that in the `initState`, that way we can be sure that it will be executed once.

Reply:
@rabble at least on main branch I'm not seeing now the of the redirect, so I guess we can close this PR, or did you fix something else? But I agree with @erickzanardo on the comment of not adding any callback in the build method since that could create undesired side effects